test/context/testQueueFactory.spec.js

```diff
@@ -0,0 +1,129 @@
+import testQueueFactory from '../../src/context/testQueueFactory'
+import runTest from '../../src/context/runTest'
+
+jest.mock(
+  '../../src/context/pubSubFactory',
+  () => () => {
+    const listeners = new Set()
+    return {
+      clearListeners: jest.fn(() => listeners.clear()),
+      listen: jest.fn((listener) => {
+        listeners.add(listener)
+        return () => listeners.delete(listener)
+      }),
+      notify: jest.fn((...args) => listeners.forEach(ls => ls(...args))),
+    }
+  }
+)
+
+jest.mock(
+  '../../src/context/runTest',
+  () => jest.fn(() => Promise.resolve({ pass: true }))
+)
+
+function delay(time = 0) {
```

Comment:
The mock promise return the object to emulate the expected result and have test on line 72 to behave as expected.
Test name is : 'should notify listeners when a test return its result'. I can modify the test to reflect an empty value returned from the promise but I'm not convinced.